File: Python/PE-3.py
# https://projecteuler.net/problem=3


def max_prime(n):
    d = 2
    while n > 1:
        while n % d == 0:
            n = n // d  # n is integer
        d += 1
    return d - 1


def main():
    print(max_prime(600851475143))


if __name__ == '__main__':
    main()

File: Go/PE-3/PE-3.go
// https://projecteuler.net/problem=3

package main

import "fmt"

func max_prime(number int) int {
	var d int = 2
	for number > 1 {
		for number % d == 0 {
			number = number / d
		}
		d++
	}
	d--
	return d
}

func main() {
	fmt.Println(max_prime(600851475143))
}
